core/headerchain.go

```diff
@@ -463,6 +466,78 @@ func (hc *HeaderChain) SetCurrentHeader(head *types.WorkObject) error {
 			"Number": head.NumberArray(),
 		}).Info("Setting the current header")
 		hc.currentHeader.Store(head)
+		if hc.config.DuplicateUTXOSet && nodeCtx == common.ZONE_CTX && hc.ProcessingState() {
+			_, order, err := hc.engine.CalcOrder(hc, head)
+			if err != nil {
+				hc.logger.Errorf("Unable to duplicate: Error calculating order for block %s: %+v", head.Hash(), err)
+			}
+			if order == common.PRIME_CTX {
+				primeBlockCounter++
```

Comment:
for now we should do this based on the prime number i think, because if we have counter and its not resilient to uncles